src/solve.jl

```diff
@@ -90,8 +95,10 @@ function __solve(prob::OptimizationProblem, opt, data = DEFAULT_DATA;cb = (args.
 
 	@withprogress progress name="Training" begin
 	  for (i,d) in enumerate(data)
-		gs = prob.f.adtype isa AutoFiniteDiff ? Array{Number}(undef,length(θ)) : DiffResults.GradientResult(θ)
-		f.grad(gs, θ, d...) 
+	    gs = Flux.Zygote.gradient(ps) do
```

Comment:
Hmm, don't we want to allow other AD packages to be used with Flux optimisers?

Reply:
We want to. But the implementation was completely broken downstream so I reverted for now until we have a solution.